web/modules/mof/src/ModelViewBuilder.php

```diff
@@ -229,7 +232,84 @@ private function buildComponentList(int $class, array $evaluation, string $statu
 
     foreach ($components as $component) {
       $license = $evaluation[$class]['licenses'][$component->id] ?? null;
-      $build["{$status}_components"]['#items'][] = $component->name . ($license ? " [{$license}]" : '');
+
+      //Gets license and name URLs from the model array, null if not available.
+      $license_url = $model['license_data'][0]['licenses']['components'][$component->id]['license_path'] ?? null;
```

Comment:
Getting the license is a bit trickier than that because it can be specified at several levels. This only covers the case where it is specified at the component level. You also need to account for the cases where it is specified at a higher level and inherited. See how it's done in the [ModelEvaluator](https://github.com/lfai/model_openness_tool/blob/a8d9c2ebf17fd6a62f082463a2a0c9fe913d41c0/web/modules/mof/src/ModelEvaluator.php#L76) and the [resolveLicense function](https://github.com/lfai/model_openness_tool/blob/a8d9c2ebf17fd6a62f082463a2a0c9fe913d41c0/web/modules/mof/src/ModelEvaluator.php#L171) it is calling.

Reply:
To test this out: Click on 'Evaluate Model', enter a dummy model name, click on Global licenses. Select 'This model has a global/distribution-wide license' and enter 'mylicense', and 'https://example.com/mylicense'. Click "Code components' and select 'Model architecture'. Click 'Evaluate' to submit the data and see the result.
The component 'Model architecture' is displayed along with the license 'mylicense' but the link is missing.
It should also work if you enter a license via 'This model has a global license for code components' under 'Global licenses'.